### PR TITLE
Show lead form for selected file

### LINKED ISSUE #1856: Bulk Upload: Create form and preview pane
### New Addition
* Make form for selected lead (maybe reuse form created in add a source popup)
* Make preview bar in the bottom

### Attachments
- [Figma Link](https://www.figma.com/proto/HaQd4COVlpSMoqSfgL5ZDx/DEEP_UI_LIBRARY-%5Bin-progress%5D?node-id=6887%3A11402&scaling=min-zoom&hide-ui=1)

### PR DESCRIPTION
**Pull request by samshara**
- fixes #1856

## Changes

* Show selected file in lead form

- [x] temporary files, auto-generated files or secret keys
- [x] build works
- [x] eslint issues
- [x] typescript issues
- [x] `console.log` meant for debugging
- [x] typos
- [x] unwanted comments
- [x] conflict markers

## This PR contains valid:

- [x] permission checks
- [ ] translations

### CODE REVIEW

File: src/newViews/Tagging/Sources/BulkUpload/Upload/FilesUpload/styles.scss [hunk 1]
```diff
@@ -6,6 +6,8 @@
         display: flex;
         flex-direction: column;
         flex-grow: 1;
+        border: none;
+        --height-border: 0;
```
Comment (tnagorra) [inline src/newViews/Tagging/Sources/BulkUpload/Upload/FilesUpload/styles.scss:10]: Is this used?

File: src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/index.tsx [hunk 2]
```diff
@@ -32,15 +33,22 @@ function FilesUploaded(props: Props) {
         files = [],
     } = props;
 
+    const [selectedFileId, setSelectedFileId] = useState<number | undefined>();
+    const [searchText, setSearchText] = useState<string | undefined>();
+
+    const handleSelection = useCallback((id: number) => {
+        setSelectedFileId(id);
+    }, []);
```
Comment (tnagorra) [inline src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/index.tsx]: Why create this separate function

File: src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/index.tsx [hunk 3]
```diff
@@ -51,34 +59,48 @@ function FilesUploaded(props: Props) {
         return files;
     }, [files, searchText]);
 
+    const handleTextChange = useCallback((value: string | undefined) => {
+        setSearchText(value);
+    }, []);
```
Comment (tnagorra) [inline src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/index.tsx]: Why create this separate function

File: src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/LeadEdit/index.tsx [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        publishedOn: formatDateToString(new Date(), 'yyyy-MM-dd'),
+        confidentiality: 'unprotected',
+        isAssessmentLead: false,
+        priority: 100,
+        attachment: {
+            id: file?.id,
+            file: file?.file,
+            mimeType: file?.mimeType,
+            title: file?.title,
+        },
```
Comment (tnagorra) [inline src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/LeadEdit/index.tsx]: Why not just pass file?

Comment (tnagorra) [reply src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/LeadEdit/index.tsx]: IDK if attachment should be an object in PartialForm

File: src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/LeadEdit/index.tsx [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    const {
+        value,
+        setValue,
+        setFieldValue,
+        error: riskyError,
+    } = useForm(schema, partialLead);
+
+    useEffect(() => {
+        setValue(partialLead);
+    }, [partialLead, setValue]);
```
Comment (tnagorra) [inline src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/LeadEdit/index.tsx:66]: Why doesn't LeadEditForm handle form handling?

**Review by tnagorra — CHANGES_REQUESTED** [review]

File: src/newViews/Tagging/Sources/BulkUpload/Upload/FilesUpload/styles.scss (hunk 1 above)
Comment (AdityaKhatri) [reply src/newViews/Tagging/Sources/BulkUpload/Upload/FilesUpload/styles.scss:10]: Yes, this is for overwriting the border of input under this container.
@frozenhelium not sure if I should do this.

File: src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/LeadEdit/index.tsx (hunk 4 above)
Comment (AdityaKhatri) [reply src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/LeadEdit/index.tsx]: It should not be a part of PartialForm.
I've removed it from PartialForm behavior in another PR.

File: src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/LeadEdit/index.tsx (hunk 5 above)
Comment (AdityaKhatri) [reply src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/LeadEdit/index.tsx:66]: To separate out save functionality, LeadEditForm is controlled component.

File: src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/index.tsx (hunk 2 above)
Comment (samshara) [reply src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/index.tsx]: @tnagorra react warns when it is passed directly

File: src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/index.tsx (hunk 3 above)
Comment (samshara) [reply src/newViews/Tagging/Sources/BulkUpload/FilesUploaded/index.tsx]: @tnagorra same as above

**Pull request merged**